src/main/java/javax/jmdns/impl/DNSRecord.java

```diff
@@ -683,36 +658,35 @@ protected void toString(final StringBuilder sb) {
             }
             sb.append('\'');
         }
-
     }
 
     /**
      * Service record.
      */
     public static class Service extends DNSRecord {
-        private final int     _priority;
-        private final int     _weight;
-        private final int     _port;
-        private final String  _server;
+        private final int priority;
+        private final int weight;
+        private final int port;
+        private final String server;
 
         public Service(String name, DNSRecordClass recordClass, boolean unique, int ttl, int priority, int weight, int port, String server) {
             super(name, DNSRecordType.TYPE_SRV, recordClass, unique, ttl);
-            this._priority = priority;
-            this._weight = weight;
-            this._port = port;
-            this._server = server;
+            this.priority = priority;
+            this.weight = weight;
+            this.port = port;
+            this.server = server;
```

Comment:
```suggestion
            this.server = server != null ? server : "";
```
may be a simply solution

Reply:
I've applied this suggestion on the commit (amended), I believe this has more benefit, since it could prevent NPE in other methods as well (e.g. in `write()` and `toByteArray()` method of `Service` class)